nextcord/automod.py

```diff
@@ -0,0 +1,351 @@
+        self.type: ActionType = try_enum(ActionType, data["type"])
+        self.notify_channel_id: Optional[int] = None
+        self.timeout_seconds: Optional[int] = None
+        if data.get("metadata"):
+            self._unpack_metadata(data["metadata"])  # type: ignore
+
+    def __repr__(self):
+        attrs = (
+            ("type", self.type),
+            ("notify_channel_id", self.notify_channel_id),
+            ("timeout_seconds", self.timeout_seconds),
+        )
+        inner = " ".join("%s=%r" % t for t in attrs)
+        return f"<AutoModerationAction {inner}>"
+
+    def _unpack_metadata(self, action_metadata: ActionMetadataPayload):
+
+        if action_metadata.get("channel_id") is not None:
+            self.notify_channel_id: Optional[int] = int(action_metadata.get("channel_id"))
+        if action_metadata.get("duration_seconds") is not None:
+            self.timeout_seconds: Optional[int] = action_metadata.get("duration_seconds")
+
+
+class AutoModerationRule(Hashable):
+    """Represents an auto moderation rule.
+
+    .. container:: operations
+        .. describe:: x == y
+            Check if two auto moderation rule are equal.
+        .. describe:: x != y
+            Check if two auto moderation rule are not equal.
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The auto moderation rule's ID.
+    guild: :class:`Guild`
+        The guild that this auto moderation rule active in.
+    name: :class:`str`
+        The name of this auto moderation rule.
+    event_type: :class:`EventType`
+        The type of the event that the rule should execute when this event is called.
+    trigger_type: :class:`TriggerType`
+        The trigger type of this auto moderation rule.
+    guild_id: :class:`int`
+        The guild ID that this auto moderation rule active in.
+    creator_id: :class:`int`
+        The creator of this auto moderation rule's ID.
+    enabled: :class:`bool`
+        Whether is this rule enabled or not.
+    exempt_role_ids: List[:class:`int`]
+        A list of roles that will not be affected by this rule.
+        .. note::
+            Bots are always not affected by any rule.
+    exempt_channel_ids: List[:class:`int`]
+        A list of channels that will not be affected by this rule.
+    keyword_filters: Optional[List[str]]
+        The custom filter for this auto moderation rule. `None` if not set.
+    presets: Optional[List[:class:`KeywordPresetType`]]
+        The pre-set type of this auto moderation rule. `None` if not set.
+    actions: List[:class:`AutoModerationAction`]
+        The actions that this auto moderation rule will execute if triggered.
+    """
+
+    def __init__(self, *, state: ConnectionState, guild: Guild, data: AutoModerationRulePayload):
+        self.id: int = int(data["id"])
+        self.guild: Guild = guild
+        self._state: ConnectionState = state
+        self._from_payload(data)
+
+    def __int__(self):
```

Comment:
May I inquire as the need for this?

Reply:
consistent with other stuff
`int(member)` does exist for example

Reply:
```py
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'Member'
```
seemingly not

Reply:
wtf, it used to be iirc

Reply:
it has only existed in enhanced discord.py